crates/holochain_state/src/integrate.rs

```diff
@@ -23,7 +22,10 @@ pub async fn authored_ops_to_dht_db(
     let mut should_hold_hashes = Vec::new();
 
     for (op_hash, basis) in hashes {
-        if network.authority_for_hash(basis).await? {
+        if storage_arcs
+            .iter()
+            .any(|arc| arc.contains(basis.get_loc().as_u32()))
```

Comment:
This is the crux of the change. We need to be able to check if we're supposed to be holding DHT data that we've authored. Really, this probably doesn't belong here. The logic of what belongs where should be contained within the workflows ideally.

At least for now though, does this seem like a reasonable API? We're requiring lots of things to know about the `kitsune2_api`, which I think should be okay. As long as we aren't requiring anything to know internal details of K2